cookbooks/aws-parallelcluster-config/recipes/aws_cli.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+# Recipe:: aws_cli
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Users to configure the AWS CLI for.
+# We configure the AWS CLI for all the users that are created by ParallelCluster and must interact with AWS.
+users = node["cluster"]["head_node_imds_allowed_users"]
+
+# AWS CLI configurations to set.
+# Note: only the configurations with a not empty value will be set.
+region = node["cluster"]["region"]
+configurations = {
+  "ca_bundle" =>  region.start_with?("us-iso") ? "/etc/pki/#{region}/certs/ca-bundle.pem" : "",
+}
+
+def set_aws_cli_configuration(config_name, expected_value, user)
+  execute "Setting AWS CLI configuration for user #{user}: #{config_name} set to '#{expected_value}'" do
+    user user
+    login true
+    command "aws configure set #{config_name} #{expected_value}"
+  end
+end
+
+users.each do |user|
```

Comment:
is there a reason why this is set at per user level and not at global os level?
I don't get the connection between the IDMS user Vs the user for which you are setting the ca_bundle into the AWS  configuration. Wasn't this setting supposed to be useful for every user in the system? I'm think about AD users for example, or any other custom user created

Reply:
I completely agree with you: the configuration should be for every present and future system user.
Nevertheless, the AWS CLI does not provide a way to configure something "globally" using 'aws configure'.
A way to set a "global" configuration is to set the corresponding environment variables for every system users.
I've explored and tested this approach: it works, but I've excluded it because:
1. relying on environment variables is more error prone and less speaking than using dedicated configuration commands:
2. it requires changes to user profiles and sudoers settings, to cover system users.
3. it requires changes to the PAM scripts executed to set the home directory up.

Regarding AD users, we can still rely on the aws configure approach by including in this PR the step (3).
I've preferred to exclude to cover AD users because the customer is responsible for them.